examples/getting_started/quickstart.ipynb

```diff
@@ -0,0 +1,455 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Mirascope Quickstart Guide\n",
+    "\n",
+    "1. [Setup](#Setup)\n",
+    "2. [Prompt Templates](#Prompt-Templates)\n",
+    "3. [Basic LLM Call](#Basic-LLM-Call)\n",
+    "4. [Streaming Responses](#Streaming-Responses)\n",
+    "5. [Response Models](#Response-Models)\n",
+    "6. [Asynchronous Processing](#Asynchronous-Processing)\n",
+    "7. [JSON Mode](#JSON-Mode)\n",
+    "8. [Output Parsers](#Output-Parsers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "Mirascope supports various LLM providers, including [OpenAI](https://openai.com/), [Anthropic](https://www.anthropic.com/), [Mistral](https://mistral.ai/), [Gemini](https://gemini.google.com), [Groq](https://groq.com/), [Cohere](https://cohere.com/), [LiteLLM](https://www.litellm.ai/), [Azure AI](https://azure.microsoft.com/en-us/solutions/ai), and [Vertex AI](https://cloud.google.com/vertex-ai). For the purposes of this guide, we will be using OpenAI. Let's start by installing Mirascope and its dependencies:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install \"mirascope[openai]\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": "This command installs Mirascope along with the necessary packages for the OpenAI integration."
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "os.environ[\"OPENAI_API_KEY\"] = \"YOUR_API_KEY\"\n",
+    "# Set the appropriate API key for the provider you're using"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prompt Templates\n",
+    "\n",
+    "Prompt templates in Mirascope allow you to create dynamic and reusable prompts."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mirascope.core import prompt_template\n",
+    "\n",
+    "\n",
+    "@prompt_template(\"What is the capital of {country}?\")\n",
+    "def get_capital_prompt(country: str): ...\n",
+    "\n",
+    "\n",
+    "print(get_capital_prompt(\"Japan\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example:\n",
+    "1. We import the `prompt_template` decorator from Mirascope.\n",
+    "2. We define a function `get_capital_prompt` and decorate it with `@prompt_template`.\n",
+    "3. The `{country}` placeholder in the template will be replaced with the actual argument passed to the function.\n",
+    "4. When we call `get_capital_prompt(\"Japan\")`, it returns the string \"What is the capital of Japan?\"\n",
```

Comment:
I think we don't need the example. because, the output is `None`.
But, we should explain about `@prompt_template`. 🤔

Reply:
Should we change the order of the prompt and call sections?
I think we can delay the explanation in the call section (e.g., "We can use @prompt_template to define prompt templates. We will explain the details of @prompt_template in the next section.")

Reply:
hmmm, i like the idea of `prompt_template` actually turning the function into something useful, this way we can follow a more natural flow of explanation (and it's easier to explain `prompt_template` because we can run the functions and show what happens so it's more clear)

What if we had a `prompt_template` decorated function always return the list of `BaseMessageParams` including any computed fields from the dynamic config but ignore things like tools, metadata, and messages (particularly messages because if a user uses messages with a call decorator then the prompt template is ignored anyway and shouldn't be used).

Then, inside of the decorator for each of the paths we can check if the function being decorated has something like `_original_fn` attached to it, grab the dynamic config from there, if it has messages, use those, and if not, call the decorated function to get the list of messages (which could all happen inside of the base setup call).

WDYT?